src/batches/setBatchGZip.m

```diff
@@ -0,0 +1,22 @@
+% (C) Copyright 2020 CPP BIDS SPM-pipeline developers
+
+function matlabbatch = setBatchGZip(unzippedNiifiles, keepUnzippedNii)
+  %
+  % Set the batch for GZip the 4D volumes
+  %
+  % USAGE:: 
+  %
+  %   matlabbatch = setGZip(unzippedNiifiles, deleteUnzippedNii)
+  %
+  % :param unzippedNiifiles: List of volumes to be gzipped
+  % :type unzippedNiifiles: array
+  % :param keepUnzippedNii: Obligatory argument. Boolean to decide to delete the unzipped files
+  % :type keepUnzippedNii: boolean
+  %
+  % :returns: - :matlabbatch: (struct) The matlabbath ready to run the spm job
+
+  matlabbatch{1}.cfg_basicio.file_dir.file_ops.cfg_gzip_files.files = unzippedNiifiles;
```

Comment:
Something we might have to decide for all functions that create batches.

That they simply "extend" a previous batch that is passed to them as argument:
```
matlabbatchOUT = setBatchFoo(matlabbatchIN, arg1)
```

where `numel(matlabbatchOUT) = numel(matlabbatchIN)  + 1`

The reason why this default way of setting up is that it makes easy to "chain" `setBatch` functions to create a list of batches to then be executed in one single workflow.

The template could look like that.


```
function matlabbatch = setBatchFoo(matlabbatch, argin2, argin3)
  %
  % Short description of what the function does goes here.
  %
  % USAGE::
  %
  %   [argout1, argout2] = setBatchFoo([matlabbatch] [, argin2 == default] [, argin3])
  %
  % :param matlabbatch: 
  % :type matlabbatch: structure
  % :param argin2: optional argument and its default value. And some of the
  %               options can be shown in litteral like ``this`` or ``that``.
  % :type argin2: string
  % :param argin3: (dimension) optional argument
  %
  % :returns:
  % 
  %           - :matlabbatch: (structure)
  %

  printBatchName('nameOfThisBatch');

  if nargin<1
   matlabbatch = [];
  end

  matlabbatch{end + 1}. %
  matlabbatch{end}. % 

end
```

Reply:
Another advantage in terms of coding is that it helps to know "Ah it is a `setBatch` function so the first argument is always `matlabbatch`".

Reply:
let me out this in an issue so we remember